apps/dav/lib/CardDAV/CardDavBackend.php

```diff
@@ -850,6 +852,8 @@ public function deleteCard($addressBookId, $cardUri) {
 	 * @return array
 	 */
 	public function getChangesForAddressBook($addressBookId, $syncToken, $syncLevel, $limit = null) {
+		$maxLimit = $this->config->getSystemValueInt('carddav_sync_request_truncation', 50000);
```

Comment:
```suggestion
		$maxLimit = $this->config->getSystemValueInt('carddav_sync_request_truncation', 2500);
```

Please let's use a lower value to reduce the amount of work the remote has to do.